x/migration/types/message_recover_morse_account_test.go

```diff
@@ -3,40 +3,54 @@ package types
 import (
 	"testing"
 
+	cometcrypto "github.com/cometbft/cometbft/crypto/ed25519"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/stretchr/testify/require"
 
 	"github.com/pokt-network/poktroll/testutil/sample"
 )
 
 func TestMsgRecoverMorseAccount_ValidateBasic(t *testing.T) {
+	validMorseSrcAddress := cometcrypto.GenPrivKey().PubKey().Address().String()
 	tests := []struct {
-		name string
-		msg  MsgRecoverMorseAccount
-		err  error
+		name          string
+		msg           MsgRecoverMorseAccount
+		expectedError error
 	}{
 		{
-			name: "invalid address",
+			name: "valid message",
 			msg: MsgRecoverMorseAccount{
-				Authority: "invalid_address",
+				Authority:          sample.AccAddress(),
+				ShannonDestAddress: sample.AccAddress(),
+				MorseSrcAddress:    validMorseSrcAddress,
 			},
-			err: sdkerrors.ErrInvalidAddress,
-		}, {
-			name: "valid address",
+		},
+		{
+			name: "invalid authority address",
```

Comment:
Do you mind adding a test for a valid address but incorrect authority?

Reply:
Not sure this could go into basic validation, which only handles intrinsic checks.

We could hard code `authtypes.NewModuleAddress(govtypes.ModuleName).String()` as the only valid authority address though :thinking:

Reply:
Noted - nvm